history-service/model/repository.js

```diff
@@ -0,0 +1,21 @@
+import HistoryModel from './history-model.js';
+import 'dotenv/config';
+
+//Set up mongoose connection
+import mongoose from 'mongoose';
+
+let mongoDB = process.env.ENV == "PROD" ? process.env.CLOUD_DB_URL : process.env.LOCAL_DB_URL;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+
+export async function getQuestionHistory(userName) {
+    return HistoryModel.find({ userName: userName }).select('title topic difficulty -_id');
```

Comment:
If we only want to display distinct questions, we can add a distinct clause and order by the record creation time in descending order to allow users to view latest question solved first.

Reply:
👍